plugins/woocommerce-admin/client/activity-panel/panels/setup-tasks/setup-tasks-panel.tsx

```diff
@@ -12,7 +12,7 @@ type QueryTypeProps = {
 export const SetupTasksPanel = ( { query }: QueryTypeProps ) => {
 	return (
 		<div className="woocommerce-setup-panel">
-			<Tasks query={ query } />
+			<Tasks query={ query } context="sidebar" />
```

Comment:
Good call, seems my subconscious was telling me the same in my naming contentions. I've made this change in https://github.com/woocommerce/woocommerce/pull/33287/commits/2d69ee33272a64c29a252f8b42b06eeee7facb46.

Reply:
I'm a little torn on this. My thinking was that this particular context provider would serve the tasks feature and all of it's sub-components, no matter where the `<Tasks />` component is, be that the homescreen or sidebar. It's also meant to be a bit more general, so it could be expanded to include properties in addition to the current `context` property that I added.

With that in mind it makes sense to be just to embed it in the root `<Tasks />` component as I've done here to avoid having to encapsulate every instance of that component with the provider. As it is we do have the one optional `context` prop so that I can relay that into the provider (it's confusing that there are two contexts in this conversation, so I could rename the property potentially). But having that one top-level prop feels simpler to me than the alternative of multiple instances of the provider.

If we move it up to encapsulate the sidebar, I suppose it just becomes `<SidebarContext />` ? That would work to relay to any sub-component within the sidebar that it's in the sidebar, but I haven't seen that need outside of Tasks currently. That and it wouldn't be at all useful to the instance of the Tasks component on the homescreen.

Does that make sense @joshuatf ?

Reply:
So @joshuatf , I was contemplating pushing back on this a bit, but a came across the fact that [you can nest the same provider within itself to override the value](https://medium.com/@NickIannelli/nested-context-the-underrated-aspect-thats-probably-missing-from-your-react-app-16e73f7d1), which made this click for me. 

This type of solution seems more useful in practice on a broader scale, so I created a `LayoutContext` within the root layout of our application which should give use the ability to see "where we are" on the page anywhere in the application. It's initialized using the page name (derived from the breadcrumbs), but it can be extended easily as I'm doing in the activity-panel. 

I may have gotten slightly carried away with this. Let me know what you think!